consensus/scheme/rolldpos/rolldposctx.go

```diff
@@ -154,6 +157,44 @@ func (ctx *rollDPoSCtx) MintBlock() (consensusfsm.Endorsement, error) {
 	if blk == nil {
 		actionMap := ctx.actPool.PendingActionMap()
 		log.L().Debug("Pick actions from the action pool.", zap.Int("action", len(actionMap)))
+		if ctx.electionCommittee != nil {
+			// prepare next epoch delegates
+			if ctx.round.height == ctx.epoch.height+uint64(ctx.cfg.NumDelegates)*uint64(ctx.cfg.NumSubEpochs)/2 {
+				blkTime, err := ctx.getBlockTime(ctx.epoch.height)
+				if err != nil {
+					return nil, err
+				}
+				beaconHeight, err := ctx.electionCommittee.HeightByTime(blkTime)
+				if err != nil {
+					return nil, err
+				}
+				r, err := ctx.electionCommittee.ResultByHeight(beaconHeight)
+				if err != nil {
+					return nil, err
+				}
+				l := state.CandidateList{}
+				for _, c := range r.Delegates() {
+					l = append(l, &state.Candidate{
+						Address:       string(c.OperatorAddress()),
+						Votes:         c.Score(),
+						RewardAddress: string(c.RewardAddress()),
+					})
+				}
+				// TODO: (zhi) some way to get producer nonce
+				nonce := ctx.epoch.num + 1
```

Comment:
This nonce may not be a valid one. You could simply set nonce = 0, to bypass nonce check for this action